QuickStart/ViewController.swift

```diff
@@ -0,0 +1,343 @@
+        // In multi-account applications, account should be retrieved by home account identifier or username instead
+
+        
+            guard let cachedTokens = ADKeychainTokenCache.defaultKeychain().allItems(nil) else {
+                self.updateLogging(text: "Didn't find a default cache. This is very unusual.")
+                
+                return nil
+            }
+
+            if !(cachedTokens.isEmpty) {
+                
+                // In the token cache, refresh tokens and access tokens are separate cache entries.
+                // Therefore, you need to keep looking until you find an entry with an access token.
+                for (_, cachedToken) in cachedTokens.enumerated() {
+                    if cachedToken.accessToken != nil {
+                        return cachedToken
+                    }
+                }
+            }
+
+        return nil
+    }
+
+    func updateLogging(text : String) {
+
+        if Thread.isMainThread {
+            self.loggingText.text = text
+        } else {
+            DispatchQueue.main.async {
+                self.loggingText.text = text
+            }
+        }
+    }
+
+    func updateSignoutButton(enabled : Bool) {
+        if Thread.isMainThread {
+            self.signoutButton.isEnabled = enabled
+        } else {
+            DispatchQueue.main.async {
+                self.signoutButton.isEnabled = enabled
+            }
+        }
+    }
+    
+    /**
+        This button will invoke the call to the Microsoft Graph API. It uses the
+        built in URLSession to create a connection.
+     */
+
+    func callAPI(retry: Bool = true) {
+
+        // Specify the Graph API endpoint
+        let url = URL(string: kGraphURI + "/v1.0/me/")
+        var request = URLRequest(url: url!)
+        
+        guard let accessToken = currentAccount()?.accessToken else {
+            // We haven't signed in yet, so let's do so now, then retry.
+            // To ensure we don't prompt the user twice,
+            // we set retry to false. If acquireToken() has some
+            // other issue we don't want an infinite loop.
+            
+            if retry {
+                
+                self.acquireToken() { (success) -> Void in
+                    if success {
+                        self.callAPI(retry: false)
+                    }
+                }
+            } else {
+                
+                self.updateLogging(text: "Couldn't get access token and we were told to not retry.")
+            }
+            return
+        }
+    
+        // Set the Authorization header for the request. We use Bearer tokens, so we specify Bearer + the token we got from the result
+        request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
+        
+        dataTask = defaultSession.dataTask(with: request) { data, response, error in
+            defer { self.dataTask = nil }
+            
+            if let error = error {
```

Comment:
Here I don't think we need to check this.
Check here, with status code check from 200 to 401.

https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/dev/Samples/ios/SampleAppiOS-Swift/SampleAppiOS-Swift/SampleGraphRequest.swift#L80

Reply:
So I took a different approach and explicitly stated what codes I'm looking for and doing work based on those codes. My thinking is the approach of this sample should be to show the flow and some of the concepts to watch out for. 200 is success, 401 is unauthorized, and respond to them in kind. Certainly, a developer would want to catch all not 200 and throw an error, but I feel like being explicit here helps them understand the different retry concepts and what the server is doing. It's why I chose verbose "retry" instead of a completion() for callAPI(). I think callbacks in general look like magic to Level 100 developers, and certainly some may come to us without knowing HTTP codes. So.. 

However, to balm the wound and save face, I did add a comment to the top of the file that says "this is crap code to show you concepts. If you want a real sample to cut/paste from, use the multi-file, fulling polymorphic, functionally written, error code laden, super sample from @unpluggedk  in the main repo :-)

Thoughts?

Reply:
Sure, I see where this approach comes from.
The only small gripe I have here is the repeated casting of response, but that doesn't mean your approach is wrong

Reply:
and this really shouldn't happen, but what if response is not a urlresponse :p